File: app/src/main/java/at/shockbytes/dante/util/ResourceManager.java
package at.shockbytes.dante.util;

import android.content.Context;
import android.content.Intent;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

import at.shockbytes.dante.R;
import at.shockbytes.dante.util.books.Book;

/**
 * @author Martin Macheiner
 *         Date: 30.04.2017.
 */

public class ResourceManager {

    private static SimpleDateFormat SDF = new SimpleDateFormat("dd. MMM yyy - kk:mm",
                                                                Locale.getDefault());


    public static String formatTimestamp(long timeMillis) {
        return SDF.format(new Date(timeMillis));
    }

    public static Intent createSharingIntent(Context c, Book b) {

        String msg = c.getString(R.string.share_template, b.getTitle(), b.getGoogleBooksLink());
        return new Intent()
                .setAction(Intent.ACTION_SEND)
                .putExtra(Intent.EXTRA_TEXT, msg)
                .setType("text/plain");
    }

}

File: app/src/main/java/at/shockbytes/dante/util/ResourceManager.kt
package at.shockbytes.dante.util

import android.content.Context
import android.content.Intent
import at.shockbytes.dante.R
import at.shockbytes.dante.util.books.Book
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author Martin Macheiner
 * Date: 30.04.2017.
 */

object ResourceManager {

    fun formatTimestamp(timeMillis: Long): String {
        return SimpleDateFormat("dd. MMM yyy - kk:mm", Locale.getDefault())
                .format(Date(timeMillis))
    }

    fun createSharingIntent(c: Context, b: Book): Intent {

        val msg = c.getString(R.string.share_template, b.title, b.googleBooksLink)
        return Intent()
                .setAction(Intent.ACTION_SEND)
                .putExtra(Intent.EXTRA_TEXT, msg)
                .setType("text/plain")
    }

}
